bika/lims/browser/analysisrequest/publish.py

```diff
@@ -642,19 +650,29 @@ def getAnaysisBasedTransposedMatrix(self, ars):
                             # here - service fields are all inside!
                             'service': an,
                             'accredited': an.getAccredited(),
-                            'ars': {ar.id: an.getFormattedResult()}
+                            'ars': {ar.id: an.getFormattedResult()},
                         }
                     }
                 elif an_title not in analyses[cat]:
                     analyses[cat][an_title] = {
                         'service': an,
                         'accredited': an.getAccredited(),
-                        'ars': {ar.id: an.getFormattedResult()}
+                        'ars': {ar.id: an.getFormattedResult()},
                     }
                 else:
                     d = analyses[cat][an_title]
                     d['ars'][ar.id] = an.getFormattedResult()
                     analyses[cat][an_title] = d
+
+                interims = hasattr(an, 'getInterimFields') \
+                    and an.getInterimFields() or []
+                analyses[cat][an_title]['interims'] = {}
+                if len(interims) > 0:
+                    an_interims_report = [a for a in interims if 'report' in a]
+                    interims_per_ar.update({ar.id: an_interims_report})
+                    analyses[cat][an_title]['interims'] = interims_per_ar
```

Comment:
I think you need to check here if the interim's "report" setting is True, otherwise it will always display all interims. Something similar to could work:
```python
interims = analysis.getInterimFields()
an_interims_report = filter(lambda interim: interim.get('report', False), interims)
interims_per_ar.update({ar.id: an_interims_report})
analyses[cat][an_title]['interims'] = interims_per_ar
```
Note that there is no need to use `hasattr`, cause [`InterimFields` is defined in `AbstractAnalysis`](https://github.com/senaite/senaite.core/blob/master/bika/lims/content/abstractanalysis.py#L121-L135), so is always available, regardless of the analysis type (`DuplicateAnalysis`, `ReferenceAnalysis`, `Analysis`, etc.)